backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -125,6 +129,15 @@ public void deleteUser(UUID uuid) {
 
   @Override
   public UserResponse getUser(UUID uuid) {
+
+    LoggedUserDto loggedUserDto = currentUserService.getLoggedUser();
+    String loggedUserRole = new ArrayList<>(loggedUserDto.getAuthorities()).get(0).toString();
```

Comment:
You can create a method in LoggedUserDto that will accept userRole and the logic will be to check if user has that role.. Something like **_hasRole(String role)_** or whatever type you need as input param. That way that logic will be reusable, this code will be there and you can call that method with `loggedUserDto.hasRole(RoleEntity.ROLE_WORKER.name()) `

Reply:
Yes, that would be nice. Will do something like that